### PR TITLE
[WIP][Enhancement] Support mapping varlen or long length primary keys into 128bit hash

### LINKED ISSUE #3940: Support mapping varlen or long length primary keys into 128bit hash in PersistentIndex
This issue proposes the ability to store 128bit hashes instead of actual primary key binary string in PrimaryIndex, as a configurable property. Good quality 128bit hash should be `practical` to eliminate collisions for typical tablet primary index use cases. 
Mapping varlen/long primary keys into fix length hashes can save a lot of memory & IO.

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #3940

## Problem Summary(Required) ：

This pr proposes the ability to store 128bit hashes instead of actual primary key binary string in PrimaryIndex, as a configurable property. Good quality 128bit hash should be practical to eliminate collisions for typical tablet primary index use cases.

Mapping varlen/long primary keys into fix length hashes can save a lot of memory & IO.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/common/config.h [hunk 1]
```diff
@@ -738,6 +738,15 @@ CONF_Int32(max_batch_publish_latency_ms, "100");
...
 CONF_String(jaeger_endpoint, "");
 
+// Support mapping varlen or long length primary keys into 128bit hash or not
+// If true, if key columns of primary key table contain varchar type, we will use a hash value(int128) to
+// replace the unqiue encoded key which can  save a lot of memory.
+// However, use fix length hash may encounter conflicts which can lead to the loss of some duplicate data.
+// Although the probability is very small, but it is possible. Please make sure you can accept it before
+// enable this feature.
+// default: false
+CONF_Bool(enable_mapping_pk_into_hash, "false");
```
Comment (decster) [inline be/src/common/config.h:748]: better make it a per table config than per BE config

Comment (decster) [reply be/src/common/config.h:748]: also, it may cause inconsistency between replicas

Comment (decster) [reply be/src/common/config.h:748]: For simplicity at start, this property can only be set when creating a table, not allowed for modification.

**Comment by sevev** [comment]
Don't approve plz. This pr is not finished, i will improve it in near future

**Comment by sevev** [comment]
close first, this pr need to be reconsider

**Pull request closed without merging**